### DISCUSSION TITLE
How was the list of restricted characters for `ensurePath` decided?

### DISCUSSION THREAD

**Discussion by chopeen** [renovatebot/renovate#21811] [Suggest an Idea]
### Type of discussion.

I just want to chat

### Tell us more.

The original list of restricted characters was very broad, then some characters were un-restricted, but the current regex:

```
const restricted = /[#%&<>*?\b\n\r\0!'"|‘“^`]/;
```

still prohibits some characters that are valid in paths in any major OS. I personally struggle with single quotes `'`, ampersands `&` and hashes `#` as they appear in quite a few repositories handled by our Renovate instance.

The code quotes http://www.mtu.edu/umc/services/digital/writing/characters-avoid/, but this article talk about _"web folders as well as for downloadable files"_, not source code.

What's the reason why `ensurePath` was added in https://github.com/renovatebot/renovate/pull/21433?  
What was the problem that needed to be solved?

---

Related PR: https://github.com/renovatebot/renovate/pull/21433, followed by:

- https://github.com/renovatebot/renovate/pull/21476
- https://github.com/renovatebot/renovate/pull/21482
- https://github.com/renovatebot/renovate/pull/21530
- https://github.com/renovatebot/renovate/pull/21625

**Comment by viceice** [comment]
I'll remove the regex check and let the os handle invalid chars tomorrow.

it was added for more safety

**Comment by viceice** [comment]
- #21871
